UI/Pages/TuitionPartner.cshtml.cs

```diff
@@ -181,27 +167,102 @@ public QueryHandler(ILocationFilterService locationService, INtpDbContext db, IL
                 tp.LegalStatus);
         }
 
-        private async Task<IEnumerable<string>> GetTuitionTypesCovered(
-            string? localAuthorityDistrictCode,
-            Domain.TuitionPartner tp,
+        private async Task<IResult<TuitionPartnersResult>> GetTPResult(Query request, CancellationToken cancellationToken)
+        {
+            var locationResult = await GetSearchLocation(request, cancellationToken);
+
+            LocationFilterParameters location = new();
+
+            if (!locationResult.IsSuccess)
+            {
+                //Shouldn't be invalid, unless query string edited - since postcode on this page comes from previous page with validation
+                _logger.LogWarning("Invalid postcode '{Postcode}' provided on TP details page", request.Postcode);
+
+                //Set to null and contine to get nationwide data
+                request.Postcode = null;
+            }
+            else
+            {
+                location = locationResult.Data;
+            }
+
+            var tpResult = await FindTuitionPartner(
+                        location,
+                        request,
+                        cancellationToken);
+
+            if (tpResult is IErrorResult tpError)
+            {
+                return tpError.Cast<TuitionPartnersResult>();
+            }
+
+            var result = new TuitionPartnersResult(tpResult.Data, location.LocalAuthority);
+
+            return Result.Success(result);
+        }
+
+        private async Task<IResult<LocationFilterParameters>> GetSearchLocation(Query request, CancellationToken cancellationToken)
```

Comment:
This logic is repeated in a few places. Would be good to refactor into a single place.

Reply:
Yep, I have that on my todo list - possibly move it in to a base handler.  There are a few other bits that I want to refactor - can go through them with you and Kola.  But wanted to get this in to 720 so we start on the merge ASAP.